x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/index.ts

```diff
@@ -146,6 +150,16 @@ interface BaseOperationDefinitionProps<C extends BaseIndexPatternColumn> {
    * present on the new index pattern.
    */
   transfer?: (column: C, newIndexPattern: IndexPattern) => C;
+  /**
+   * if there is some reason to display the operation in the operations list
+   * but disable it from usage, this function returns the string describing
+   * the status. Otherwise it returns undefined
+   */
+  getDisabledStatus?: (indexPattern: IndexPattern) => string | undefined;
+  /**
+   * Returns true if column contains invalid references
+   */
+  hasInvalidReferences?: (column: C, indexPattern: IndexPattern) => boolean;
```

Comment:
`hasInvalidReferences` is really similar to `getErrorMessage` on the reference column Wylie introduced recently. Do we need both or does it make sense to collapse them into a single one used for both purposes?

Reply:
Yes, it makes sense, I think I started working on it before `getErrorMessage` got introduced to the codebase. I'll take a look how to merge it 🙆‍♀️